### PR TITLE
Updating the gcp consumer

### PR DESCRIPTION
**Pull request by duallain**
update users if they have previously been created in the system
point at a more correct source of truth for ldap data

### CODE REVIEW

**Comment by gcoxmoz** [comment]
Is there some styling reason for all the single-quote to double-quote changes?  My internal pylinter is having a "you don't need to do that, there's no interpolation there."
If so, I'd love to see it as a "here's the meat of the change" and "here's just quote changes" breakout because I'm having a hard time finding the meat of the change.

From what I see, it looks like it's a bucket/file change, and a first/last name being dropped, and it looks sane, but I don't trust my read so I'm commenting rather than approving.

File: gsuite_cloud_users_driver/cloud.py [hunk 1]
```diff
@@ -13,7 +14,7 @@
 class Directory(object):
     def __init__(self):
         self.auth = None
-        self.scope = 'https://www.googleapis.com/auth/admin.directory.user'
+        self.scope = "https://www.googleapis.com/auth/admin.directory.user"
```
Comment (floatingatoll) [inline gsuite_cloud_users_driver/cloud.py:17]: Is there some python styling system that's applying this change? I'd prefer to see unrelated / unnecessary changes in a separate PR, but I don't object strongly enough to block here.

File: gsuite_cloud_users_driver/cloud.py [hunk 2]
```diff
@@ -23,63 +24,85 @@ def _discover_service(self):
         store = ServiceAccountCredentials.from_json_keyfile_dict(
             keyfile_dict=self._get_keyfile_dict(), scopes=self.scope
         )
-        delegated_credentials = store.create_delegated('iam-robot@gcp.infra.mozilla.com')
+        delegated_credentials = store.create_delegated(
+            "iam-robot@gcp.infra.mozilla.com"
```
Comment (floatingatoll) [inline gsuite_cloud_users_driver/cloud.py:28]: It's confusing to see this inline rather than in the credentials dict we just loaded. Ah, but this is just reformat, I'll try to filter more carefully.

File: gsuite_cloud_users_driver/cloud.py [hunk 3]
```diff
@@ -88,6 +94,8 @@ def create(self, user_dict):
             if e.reason == "Entity already exists.":
                 email = body.pop("primaryEmail")
                 result = self.update_user(email, body)
+            else:
+                print(e)
```
Comment (floatingatoll) [inline gsuite_cloud_users_driver/cloud.py:96]: This seems like an error print that's missing supporting context, like "Unable to create entity: ${e}" or similar.

File: gsuite_cloud_users_driver/cloud.py [hunk 4]
```diff
@@ -23,63 +24,85 @@ def _discover_service(self):
...
     def create(self, user_dict):
+        email_prefix = user_dict["primary_email"].split("@")[0]
         body = {
-            'name': {
-                'givenName': user_dict['first_name'],
-                'fullName': '{} {}'.format(user_dict['first_name'], user_dict['last_name']),
-                'familyName': user_dict['last_name']
+            "name": {
+                "givenName": email_prefix[0],
+                "fullName": email_prefix,
```
Comment (bkochendorfer) [inline gsuite_cloud_users_driver/cloud.py:82]: For me this would be `bkochendorfer`, is that right?

File: gsuite_cloud_users_driver/serverless.yml [hunk 5]
```diff
@@ -16,15 +16,15 @@ provider:
   runtime: python3.9
   environment:
     CIS_ENVIRONMENT: production
-    CIS_S3_BUCKET_NAME: cis-ldap2s3-publisher-data
```
Comment (floatingatoll) [inline gsuite_cloud_users_driver/serverless.yml:19]: Is this the only functional change in this specific commit? I can't tell what's fmt and what actually needs review.

File: gsuite_cloud_users_driver/ldap.py [hunk 6]
```diff
@@ -76,31 +81,27 @@ def to_gsuite_account_structure(self):
...
 
     def _record_to_primary_email(self, user):
-        return user.get('primaryEmail').lower()
-
-    def _record_to_first_name(self, user):
-        return user.get('firstName')
-
-    def _record_to_last_name(self, user):
-        return user.get('lastName')
+        return user
```
Comment (floatingatoll) [inline gsuite_cloud_users_driver/ldap.py:104]: If we were planning to keep gcp.infra, I would worry about discarding the intent of capturing HRIS data here someday, but since we're likely to end infraGCP I'm just going to shrug and ignore it. Whatever we're doing for firefoxGCP doesn't use this code, so I don't care that we're removing functionality.

File: gsuite_cloud_users_driver/cloud.py [hunk 7]
```diff
@@ -23,63 +24,85 @@ def _discover_service(self):
...
 
     def create(self, user_dict):
+        email_prefix = user_dict["primary_email"].split("@")[0]
         body = {
-            'name': {
-                'givenName': user_dict['first_name'],
-                'fullName': '{} {}'.format(user_dict['first_name'], user_dict['last_name']),
-                'familyName': user_dict['last_name']
+            "name": {
+                "givenName": email_prefix[0],
```
Comment (bkochendorfer) [inline gsuite_cloud_users_driver/cloud.py:81]: This is just the first initial for me, but is this consistent across Mozilla? Like if another bkochendorfer (Brittany) starts would their email be `brikochendorfer`?

File: gsuite_cloud_users_driver/ldap.py [hunk 8]
```diff
@@ -10,38 +10,43 @@
...
 
     def _assume_role(self):
-        logger.info('Assuming role for pulling ldap data.')
-        client = boto3.client('sts')
+        logger.info("Assuming role for pulling ldap data.")
+        client = boto3.client("sts")
         response = client.assume_role(
             RoleArn=os.getenv(
-                'LDAP_ASSUME_ROLE_ARN',
-                'arn:aws:iam::371522382791:role/cis-gsuite-users-driver'
```
Comment (floatingatoll) [inline gsuite_cloud_users_driver/ldap.py:24]: I guess we hard-code this twice? Why is the numeric ID changing?

**Review by floatingatoll — APPROVED** [review]
I'm assuming this is already tested and works, and so this is more likely a review-in-hindsight. Nothing really says "no stop don't", so I would not object if my questions and feedback are left unresolved.

File: gsuite_cloud_users_driver/serverless.yml [hunk 9]
```diff
@@ -16,15 +16,15 @@ provider:
...
-    CIS_LDAP_JSON_FILE: ldap-full-profile.json.xz
-    LDAP_ASSUME_ROLE_ARN: arn:aws:iam::371522382791:role/cis-gsuite-users-driver
+    CIS_S3_BUCKET_NAME: cache.ldap.sso.mozilla.com
```
Comment (bkochendorfer) [inline gsuite_cloud_users_driver/serverless.yml:19]: Bucket is real and has the json file you are pointing at.

File: gsuite_cloud_users_driver/serverless.yml [hunk 10]
```diff
@@ -16,15 +16,15 @@ provider:
...
+    CIS_LDAP_JSON_FILE: ldap_users.json.xz
+    LDAP_ASSUME_ROLE_ARN: arn:aws:iam::320464205386:role/cis-gsuite-users-driver
   iamRoleStatements:
     - Effect: Allow
       Action:
         - "sts:AssumeRole"
       Resource:
-        - arn:aws:iam::371522382791:role/cis-gsuite-users-driver
+        - arn:aws:iam::320464205386:role/cis-gsuite-users-driver
```
Comment (bkochendorfer) [inline gsuite_cloud_users_driver/serverless.yml:27]: Looks real and has the right policies for what you want to do

File: gsuite_cloud_users_driver/cloud.py (hunk 7 above)
Comment (floatingatoll) [reply gsuite_cloud_users_driver/cloud.py:81]: They would, by past processes anyways, be assigned 'brkochendorfer'.

**Comment by duallain** [comment]
> Is there some styling reason for all the single-quote to double-quote changes? My internal pylinter is having a "you don't need to do that, there's no interpolation there." If so, I'd love to see it as a "here's the meat of the change" and "here's just quote changes" breakout because I'm having a hard time finding the meat of the change.

That's fair, I just ran `black .` on the whole directory.  Should have done that on a seperate git commit, but didn't so would need to choose between undoing it, and retesting everything all over again, or just living with it.  

> 
> From what I see, it looks like it's a bucket/file change, and a first/last name being dropped, and it looks sane, but I don't trust my read so I'm commenting rather than approving.

:nods: that's the intention for sure.

File: gsuite_cloud_users_driver/cloud.py (hunk 7 above)
Comment (duallain) [reply gsuite_cloud_users_driver/cloud.py:81]: the ldap file doesn't currently have first/last/full names in it.  And gcp seems to required those fields when creating a user.  So my intention with the split is to pick names that feel 'ok' to people, with the ability to edit if they so choose.  Happy to hear other ideas for what to fill in, but this seemed like a good compromise to get stuff working again.

The 'right' fix is likely to ask @gcoxmoz to have ldap send over the extra name fields, but it just doesn't feel worth it to me.

File: gsuite_cloud_users_driver/cloud.py (hunk 4 above)
Comment (duallain) [reply gsuite_cloud_users_driver/cloud.py:82]: yup, if you want to login to gcp and let me know how annoying that looks, you were one of the users I created in this iteration of the work.

File: gsuite_cloud_users_driver/ldap.py (hunk 8 above)
Comment (duallain) [reply gsuite_cloud_users_driver/ldap.py:24]: following the convention of the previous  devs, who seemed to specify a default and also always pass the argument to the lambda.  The s3 bucket was in the EIS Prod account, but is now in mozilla-iam, so assuming roles in different accounts

File: gsuite_cloud_users_driver/cloud.py (hunk 7 above)
Comment (bkochendorfer) [reply gsuite_cloud_users_driver/cloud.py:81]: haha, nope that's fine now that I understand the intent!

**Review by bkochendorfer — APPROVED** [review]
